backend/src/models/User.ts

```diff
@@ -0,0 +1,19 @@
+import mongoose, { Document } from "mongoose";
+
+export type UserDocument = {
+  name: string;
+  gender: string;
+  accountType: "admin" | "team"; // NOTE Also stored on Firebase using Custom Claims
+  approvalStatus: string; // NOTE Should this be restricted to certain values?
+  firebaseUid: string; // Linkage between firebase account and user document on MongoDb
```

Comment:
Alternatively what we could do is create a document with the same uid as the one in firebase. See this [stackoverflow post](https://stackoverflow.com/questions/23593479/insert-documents-to-specific-id-in-mongodb) on how to create a document with our own uid.